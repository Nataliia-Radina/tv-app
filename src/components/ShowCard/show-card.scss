.show-card {
  .v-image__image {
    transition: transform .15s;
  }
  .v-chip {
    position: absolute;
    top: 4px;
    right: 4px;
    z-index: 1;
  }
  a {
    text-decoration: none;
  }
  .v-card {
    position: relative;
    &:hover {
      .v-image__image {
        transform: scale(1.15);
      }
    }
  }
}
