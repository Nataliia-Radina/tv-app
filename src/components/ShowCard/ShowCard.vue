<template>
  <v-col
    cols="6"
    sm="4"
    md="3"
    lg="2"
    class="show-card"
  >
    <router-link :to="`/show/${show.show.id}`">
    <v-card tile>
      <v-chip small color="indigo" v-if="show.airtime">
        <v-icon small left>
          mdi-clock
        </v-icon>
        {{ show.airtime }}
      </v-chip>
      <v-img :src="getImageSrc(show.show.image)" height="200px" :alt="show.name"></v-img>
      <v-card-subtitle class="text-center pa-2 pb-1">
        <h4 class="text-truncate">
          {{ show.show.name }}
        </h4>
      </v-card-subtitle>
      <v-card-text align-center class="px-2 pt-0 pb-3 text-right">
        <span class="grey--text text--lighten-2">
          <v-icon small color="yellow accent-4">
            mdi-star
          </v-icon>
          {{ rating }}
        </span>
      </v-card-text>
    </v-card>
    </router-link>
  </v-col>
</template>

<script>
import './show-card.scss'
import ImageSrc from '@/mixins/ImageSrc'

export default {
  name: 'ShowCard',
  mixins: [ImageSrc],
  props: {
    show: {
      type: Object,
      required: true
    }
  },
  computed: {
    rating () {
      return this.show.show.rating.average ? this.show.show.rating.average : '-'
    }
  }
}
</script>
