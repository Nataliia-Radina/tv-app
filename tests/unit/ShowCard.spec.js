import ShowCard from '@/components/ShowCard/ShowCard'
import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
Vue.use(Vuetify)

it('expect to render ShowCard component', () => {
  const show = {
    airtime: "00:00",
    id: 1963908,
    show: {
      id: 363,
      image: { medium: "src-medium-1", original: "src-original-1" },
      name: "Show1",
      rating: { average: 5.8 }
    }
  }
  const wrapper = shallowMount(ShowCard, {
    propsData: { show }
  })
  expect(wrapper).toMatchSnapshot()
})
