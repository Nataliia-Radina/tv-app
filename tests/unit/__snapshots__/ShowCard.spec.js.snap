// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ShowCard component 1`] = `
<v-col-stub cols="6" sm="4" md="3" lg="2" tag="div" class="show-card">
  <v-card-stub loaderheight="4" tag="div" tile="true">
    <v-chip-stub color="indigo" small="true" activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete">
      <v-icon-stub small="" left="">
        mdi-clock
      </v-icon-stub>
      00:00
    </v-chip-stub>
    <v-img-stub height="200px" options="[object Object]" position="center center" src="src-original-1" transition="fade-transition"></v-img-stub>
    <v-card-subtitle-stub class="text-center pa-2 pb-1">
      <h4 class="text-truncate">
        Show1
      </h4>
    </v-card-subtitle-stub>
    <v-card-text-stub aligncenter="" align-center="" class="px-2 pt-0 pb-3 text-right"><span class="grey--text text--lighten-2"><v-icon-stub small="" color="yellow accent-4">
          mdi-star
        </v-icon-stub>
        5.8
      </span></v-card-text-stub>
  </v-card-stub>
</v-col-stub>
`;
